pandas/core/arrays/datetimelike.py

```diff
@@ -338,7 +338,6 @@ def __getitem__(
         result = cast(
             "Union[DatetimeLikeArrayT, DTScalarOrNaT]", super().__getitem__(key)
         )
-        result = super().__getitem__(key)
```

Comment:
this didn't change any tests?

Reply:
Because it is the same line as the one above (just without the cast)